opal/include/opal/sys/gcc_builtin/atomic.h

```diff
@@ -36,7 +36,7 @@
  *
  *********************************************************************/
 
-#if defined(PLATFORM_ARCH_X86_64) && defined (__GNUC__) && !defined(__llvm) && (__GNUC__ < 6)
+#if defined(PLATFORM_ARCH_X86_64) && defined(PLATFORM_COMPILER_GNU) && __GNUC__ < 8
```

Comment:
This should really be two commits (changing the default, and then changing the builtn code).  I'm happy with the platform compiler change, but why are we changing the version of GCC that has the busted MB?